src/app/products/products-search/products-search.component.ts

```diff
@@ -14,25 +14,20 @@ import { Location } from '@angular/common';
 export class ProductsSearchComponent {
   isLoading$: Observable<boolean> = this.loaderService.isLoading$;
   q = '';
-  products$: Observable<Product[]> = this.productsService.getProducts(`q=${this.q}`);
+  products$!: Observable<Product[]>;
```

Comment:
Can we write something like this instead of the code inside `constructor`?
```suggestion
  products$ = this.route.params.pipe(
    map((params) => params['q']),
    switchMap((q: string) => this.productsService.getProducts(`q=${this.q}`))
  )
```